endpoint/endpointComp/endpoint.c

```diff
@@ -47,21 +47,66 @@ static void print_help(void) {
   exit(EXIT_SUCCESS);
 }
 
+static void string_to_uint8_arr(const char* string, uint8_t arr[]) {
+  if (!string) return;
+
+  size_t str_start = 0, str_bytes = 0, arr_idx = 0;
+  // The maximum value of uint8_t is 255, hence 4 is enough for the string length including the null-terminated
+  // character.
+  char parse_str[4];
+  int parse_int;
+
+  // Seperate the string with the delimiter "," and convert the substring to uint8_t.
+  while (str_start < le_utf8_NumChars(string)) {
+    le_utf8_CopyUpToSubStr(parse_str, &string[str_start], ",", sizeof(parse_str), &str_bytes);
+    le_utf8_ParseInt(&parse_int, parse_str);
+    arr[arr_idx] = (uint8_t)parse_int;
+
+    str_start += (str_bytes + 1);
+    arr_idx++;
+  }
+  return;
+}
+
 COMPONENT_INIT {
   // FIXME:
   // The current code is a prototype for passing the CI.
   // The initialization of hardware and the input from hardware are not implemented yet.
+  int value = TEST_VALUE;
+  const char* message = TEST_MESSAGE;
+  const char* message_fmt = TEST_MESSAGE_FMT;
+  const char* tag = TEST_TAG;
+  const char* address = TEST_ADDRESS;
+  const char* next_address = TEST_NEXT_ADDRESS;
+  const char* private_key_str = NULL;
+  uint8_t private_key[32] = {0};
+  const char* device_id = TEST_DEVICE_ID;
+  const char* iv_str = NULL;
+  uint8_t iv[AES_IV_SIZE] = {0};
+
+  le_arg_SetIntVar(&value, NULL, "value");
+  le_arg_SetStringVar(&message, NULL, "msg");
+  le_arg_SetStringVar(&message_fmt, NULL, "msg-fmt");
+  le_arg_SetStringVar(&tag, NULL, "tag");
+  le_arg_SetStringVar(&address, NULL, "addr");
+  le_arg_SetStringVar(&next_address, NULL, "next-addr");
+  le_arg_SetStringVar(&private_key_str, NULL, "pvt-key");
+  le_arg_SetStringVar(&iv_str, NULL, "iv");
```

Comment:
I think initialize vector should be calculated be endpoint itself, @HowJMay what do you think?

Reply:
It should be a random byte list. I think it can be generated every time endpoint sends a message

Reply:
The implementation of reading initialization vector from command line is removed.